crates/ruff_python_formatter/src/expression/mod.rs

```diff
@@ -529,18 +529,63 @@ impl<'ast> IntoFormat<PyFormatContext<'ast>> for Expr {
 /// * The expression contains at least one parenthesized sub expression (optimization to avoid unnecessary work)
 ///
 /// This mimics Black's [`_maybe_split_omitting_optional_parens`](https://github.com/psf/black/blob/d1248ca9beaf0ba526d265f4108836d89cf551b7/src/black/linegen.py#L746-L820)
+#[allow(clippy::if_same_then_else)]
 fn can_omit_optional_parentheses(expr: &Expr, context: &PyFormatContext) -> bool {
     let mut visitor = CanOmitOptionalParenthesesVisitor::new(context);
     visitor.visit_subexpression(expr);
 
     if !visitor.any_parenthesized_expressions {
         // Only use the more complex IR when there is any expression that we can possibly split by
         false
-    } else if visitor.max_precedence == OperatorPrecedence::None {
-        true
```

Comment:
Just confirming it's intentional that the order of the branches switched here.

Reply:
Not in that it has (or should have) any behavioral change. I just tried to identify which rules apply to all (e.g. `max_precedence_count > 1` cases and which cases are "special"